app/heron/factories/concerns/contents.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Heron
+  module Factories
+    module Concerns
+      #
+      # This module adds validation and tools to manage factories that act as content in a receptacle
+      # Eg: ::Heron::Factories::Sample
+      #
+      # **Requirements**
+      # - The method or attribute content_factory needs to be defined with the class that act as
+      #   factory for contents. Eg: ::Heron::Factories::Sample
+      # - The content configuration should be already stored in @params
+      # - The content configuration object needs to be in @params[recipients_key]
+      # - The method or attribute recipients_key will need to identify the key where the config is
+      #   stored. Eg: :wells
+      # - All keys in the content configuration are considered as validated, if that is not the case,
+      #   that validation can be provided by including the module Heron::Factories::Concerns::RecipientsCoordinate
+      # - The class should include the module Heron::Factories:Concerns::CoordinatesSupport
+      #
+      # **Use**
+      # Include the module in the class after checking the previous list of requirements.
+      #
+      # **Effect**
+      # Factories for each content in each well will be generated.
+      # Any validation error from them will be aggregated in the base object.
+      #
+      # **Examples of params**
+      # 1. Empty contents
+      # {}
+      #
+      # 2. A well at position A1, with a sample that has name 'Sample 1'
+      # { wells: {'A1': {content: {name: 'Sample 1', sample_uuid: <uuid>} } } }
+      #
+      # 3. Two aliquots in position B1 from samples with name 'Sample 1' and 'Sample 2'
+      #    and both will be created under study <uuid>
+      # { wells: {'B1': {content: [{name: 'Sample 1'}, {name: 'Sample 2'}] } }, sample_uuid: <uuid>}
```

Comment:
```suggestion
      # { wells: {'B1': {content: [{name: 'Sample 1'}, {name: 'Sample 2'}] } }, study_uuid: <uuid>}
```
should the above be study_uuid?

Reply:
study_uuid, updated